modular_skyrat/modules/modular_implants/code/nifs.dm

```diff
@@ -0,0 +1,511 @@
+	if(!linked_mob || !linked_mob.blood_volume)
+		return FALSE
+
+	if(linked_mob.blood_volume < minimum_blood_level)
+		return FALSE
+
+	return TRUE
+///Preforms calibration, this is run the first time a NIF is installed in someone.
+/obj/item/organ/internal/cyberimp/brain/nif/proc/preform_calibration()
+	if(linked_mob.stat == DEAD)
+		return FALSE
+
+	if(!calibration_duration)
+		calibration_duration = world.time + calibration_time
+
+	var/percentage_done = (world.time - (calibration_duration - (calibration_time))) / calibration_time
+	switch(percentage_done)
+		if(0 to 0.1)
+			linked_mob.set_blindness(5)
+
+		if(0.2 to 0.9)
+			var/random_ailment = rand(1, side_effect_risk)
+			switch(random_ailment)
+				if(1)
+					to_chat(linked_mob, span_warning("You feel sick to your stomach"))
+					linked_mob.adjust_disgust(25)
+				if(2)
+					to_chat(linked_mob, span_warning("You feel a wave of fatigue roll over you"))
+					linked_mob.adjustStaminaLoss(50)
+
+
+		if(1 to INFINITY)
+			send_message("Installation Proccess Complete!")
+
+			calibrating = FALSE
+			is_calibrated = TRUE
+
+///This is run whenever a nifsoft is installed
+/obj/item/organ/internal/cyberimp/brain/nif/proc/install_nifsoft(datum/nifsoft/loaded_nifsoft)
+	if(broken) //NIFSofts can't be installed to a broken NIF
+		return FALSE
+
+	if((length(loaded_nifsofts) >= max_nifsofts))
+		return FALSE
+
+	if(!is_type_in_list(src, loaded_nifsoft.compatible_nifs))
+		send_message("[loaded_nifsoft.name] is incompatible with your NIF!", TRUE)
+		return FALSE
+
+	for(var/datum/nifsoft/current_nifsoft as anything in loaded_nifsofts)
+		if(loaded_nifsoft.single_install && loaded_nifsoft.type == current_nifsoft.type)
+			send_message("Multiple of [loaded_nifsoft.name] cannot be installed", TRUE)
+			return FALSE
+
+		if(current_nifsoft.type in loaded_nifsoft.mutually_exclusive_programs)
+			send_message("[current_nifsoft.name] is preventing [loaded_nifsoft.name] from being installed", TRUE)
+			return FALSE
+
+	loaded_nifsofts += loaded_nifsoft
+	loaded_nifsoft.parent_nif = src
+	loaded_nifsoft.linked_mob = linked_mob
+
+	send_message("[loaded_nifsoft.name] has been added")
+	return TRUE
+
+/obj/item/organ/internal/cyberimp/brain/nif/proc/remove_nifsoft(datum/nifsoft/removed_nifsoft, silent = FALSE)
+	if(!is_type_in_list(removed_nifsoft, loaded_nifsofts) || broken)
+		return FALSE
+
+	if(!silent)
+		send_message("[removed_nifsoft.name] has been removed", TRUE)
+
+	qdel(removed_nifsoft)
+	return TRUE
+
+/obj/item/organ/internal/cyberimp/brain/nif/proc/repair_nif(repair_amount)
+	if(durability == max_durability)
+		return FALSE
+
+	durability += repair_amount
+	if(durability > max_durability)
+		durability = max_durability
+
+	return TRUE
+
+/obj/item/organ/internal/cyberimp/brain/nif/proc/send_message(message_to_send, alert=FALSE)
+	var/datum/asset/spritesheet/sheet = get_asset_datum(/datum/asset/spritesheet/chat)
+	var/tag = sheet.icon_tag("nif-[chat_icon]")
```

Comment:
Does this work with the `Disable examine icons` lag switch?

Reply:
I have no clue about that, could you please explain this to me on discord?